### PR TITLE
Ingest Updates

### PR DESCRIPTION
**Pull request by vangheem**
### Description

- Decouple pull workers from ingest consumers
- Move to constants for stream configuration
- Reuse nats connections between consumers instead of having a unique connection for every one
- Be able to split out processed messages writes from regular transaction writes
- Decouple ingest write seq id tracking from the db transaction handling

### How was this PR tested?
Unit, integration and pain

### CODE REVIEW

**Comment by vangheem** [comment]
[sc-5275]

File: nucliadb/nucliadb/ingest/orm/processor/__init__.py [hunk 1]
```diff
@@ -277,6 +280,11 @@ async def process(
                     else None
                 )
         elif message.type == BrokerMessage.MessageType.MULTI:
+            # XXX Not supported right now
+            # MULTI, COMMIT and ROLLBACK are all not supported in transactional mode right now
+            # This concept is probably not tenable with current architecture because
+            # of how nats works and how we would need to manage rollbacks.
+            # XXX Should this be removed?
```
Comment (lferran) [inline nucliadb/nucliadb/ingest/orm/processor/__init__.py:287]: for me, we can remove all of this logic if it's not used.
I am OK if we do it in a separate PR though

File: nucliadb/nucliadb/ingest/orm/processor/__init__.py [hunk 2]
```diff
@@ -429,16 +448,18 @@ async def txn(
...
+                if transaction_check:
+                    await sequence_manager.set_last_seqid(txn, partition, seqid)
+                await txn.commit()
 
                 # Slug may have conflicts as its not partitioned properly. We make it as short as possible
-                txn = await self.driver.begin()
-                resource.txn = txn
-                await resource.set_slug()
-                await txn.commit(resource=False)
+                # XXX Does this write need to happen every time?
```
Comment (lferran) [inline nucliadb/nucliadb/ingest/orm/processor/__init__.py:456]: no it doesn't. I will put up a PR fixing this after this is merged

File: nucliadb/nucliadb/ingest/orm/processor/__init__.py [hunk 3]
```diff
@@ -429,16 +448,18 @@ async def txn(
...
+                async with self.driver.transaction() as txn:
```
Comment (lferran) [inline nucliadb/nucliadb/ingest/orm/processor/__init__.py:457]: I think we should name this transaction differently to avoid leaving the main one open?

File: nucliadb_utils/nucliadb_utils/const.py [hunk 4]
```diff
@@ -21,6 +21,37 @@ class PubSubChannels:
...
+
+    class INGEST_PROCESSED:
+        """
+        Resources that have been processed by learning need to be
+        written to the database and then Indexed.
+        """
+
+        name = "nucliadb"
+        subject = "ndb.consumer.processed"
+        group = "nucliadb-processed-messsages"
```
Comment (lferran) [inline nucliadb_utils/nucliadb_utils/const.py:42]: too many messsssssagess! 😉

File: nucliadb_utils/nucliadb_utils/transaction.py [hunk 5]
```diff
@@ -46,7 +46,12 @@ def __init__(self, uuid: str, seq: Optional[int] = None):
 
 class LocalTransactionUtility:
     async def commit(
-        self, writer: BrokerMessage, partition: int, wait: bool = False
+        self,
+        writer: BrokerMessage,
+        # same interface
```
Comment (lferran) [inline nucliadb_utils/nucliadb_utils/transaction.py:51]: is this comment needed?

File: nucliadb_utils/nucliadb_utils/transaction.py [hunk 6]
```diff
@@ -117,7 +120,11 @@ def received(waiting_for: WaitFor, event: Event, raw_data: bytes):
             pb = Notification()
             pb.ParseFromString(data)
             if pb.uuid == waiting_for.uuid and pb.action == action_type:
-                if waiting_for.seq is None or pb.seqid == waiting_for.seq:
+                if (
+                    waiting_for.seq is None
+                    or pb.seqid == waiting_for.seq
+                    or action_type != Notification.Action.INDEXED
```
Comment (lferran) [inline nucliadb_utils/nucliadb_utils/transaction.py:126]: I don't understand the reason for this or condition. Can you explain?

File: nucliadb/nucliadb/ingest/orm/processor/__init__.py (hunk 3 above)
Comment (vangheem) [reply nucliadb/nucliadb/ingest/orm/processor/__init__.py:457]: The other txn has already been committed at this point but good practice anyways!

File: nucliadb_utils/nucliadb_utils/transaction.py (hunk 6 above)
Comment (vangheem) [reply nucliadb_utils/nucliadb_utils/transaction.py:127]: If we're waiting on the resource to be indexed, the seq id will not match.
